components/brave_rewards/resources/page/reducers/wallet_reducer.ts

```diff
@@ -151,69 +148,27 @@ const walletReducer: Reducer<Rewards.State | undefined> = (state: Rewards.State,
       chrome.send('brave_rewards.setExternalWalletType', [action.payload.provider])
       break
     }
-    case types.ON_EXTERNAL_WALLET: {
+    case types.GET_EXTERNAL_WALLET: {
+      chrome.send('brave_rewards.getExternalWallet')
+      break
+    }
+    case types.ON_GET_EXTERNAL_WALLET: {
       state = { ...state }
 
-      if (action.payload.result === 24) { // type::Result::EXPIRED_TOKEN
-        chrome.send('brave_rewards.getExternalWallet')
-        break
-      }
-
-      if (action.payload.result === 9) { // type::Result::NOT_FOUND
-        state.ui.modalRedirect = 'kycRequiredModal'
-        break
-      }
-
-      if (action.payload.result === 25) { // type::Result::UPHOLD_BAT_NOT_ALLOWED
-        state.ui.modalRedirect = 'upholdBATNotAllowedModal'
-        break
-      }
-
-      if (action.payload.result === 36) { // type::Result::DEVICE_LIMIT_REACHED
-        state.ui.modalRedirect = 'deviceLimitReachedModal'
-        break
-      }
-
-      if (action.payload.result === 37) { // type::Result::MISMATCHED_PROVIDER_ACCOUNTS
-        state.ui.modalRedirect = 'mismatchedProviderAccountsModal'
-        break
-      }
-
-      if (action.payload.result === 41) { // type::Result::UPHOLD_TRANSACTION_VERIFICATION_FAILURE
-        state.ui.modalRedirect = 'walletOwnershipVerificationFailureModal'
-        break
-      }
-
-      if (action.payload.result === 43) { // type::Result::UPHOLD_INSUFFICIENT_CAPABILITIES
-        state.ui.modalRedirect = 'upholdInsufficientCapabilitiesModal'
-        break
-      }
-
-      if (action.payload.result === 44) { // type::Result::FLAGGED_WALLET
-        state.ui.modalRedirect = 'flaggedWalletModal'
-        break
-      }
+      const error = action.payload.result.error
+      switch (error) {
```

Comment:
Nit: It looks like this is switching over `number | undefined` which is fine. It might read a bit better to check for the success case first:

```js
if (action.payload.result.wallet) {
  // ...
} else {
  switch (error) {
    // ...
  }
}
```

Up to you!

Reply:
Sure, might make more sense to deal with the success case first.